File: top-interview-questions-easy/Array/best-time-to-buy-and-sell-stock-ii/122.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxProfit = 0
        length = len(prices)
        for i in range(0, length - 1):
            if (prices[i] < prices[i + 1]):
                maxProfit += prices[i + 1] - prices[i]
        return maxProfit

File: top-interview-questions-easy/Array/best-time-to-buy-and-sell-stock-ii/122.js
/**
 * @param {number[]} prices
 * @return {number}
 */
const maxProfit = function(prices) {
  const len = prices.length
  let maxProfit = 0
  for (let i = 0; i < len - 1; i++) {
    if (prices[i] < prices[i + 1]) {
      maxProfit += prices[i + 1] - prices[i]
    }
  }
  return maxProfit
}
